src/methods/to.js

```diff
@@ -619,6 +619,7 @@ export class ToImplementation {
   GetToPrimitivePureResultType(realm: Realm, input: Value): void | typeof Value {
     let type = input.getType();
     if (input instanceof PrimitiveValue) return type;
+    if (type === SymbolValue) return undefined;
     if (input instanceof AbstractValue && !input.mightBeObject()) return PrimitiveValue;
```

Comment:
Can an `AbstractValue` ever be a `Symbol` at runtime? I'd think so. But then this still might return `PrimitiveValue`. Is that okay?

Reply:
A symbol is a primitive value, so line 623 seems correct while 622 seems wrong.

Reply:
So I looked at the spec again and it looks like now the ToPrimitive operation doesn't throw. The bug is actually in BinaryExpression, because it tests for ToPrimitive being pure but it should be looking at ToString. ToString will always throw for symbols so it's not pure.